man/systemd.netdev.xml

```diff
@@ -983,6 +986,100 @@
     as the <literal>[Tun]</literal> section.</para>
   </refsect1>
 
+  <refsect1>
+    <title>[WireGuard] Section Options</title>
+
+    <para>The <literal>[WireGuard]</literal> section accepts the following
+    key:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>PrivateKey=</varname></term>
+        <listitem>
+          <para>The private key for the interface. It can be generated
+            using the <command>wg genkey</command> command. This option is
+            mandatory to use wireguard.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>ListenPort</varname></term>
+        <listitem>
+          <para>Sets UDP port for listening. Takes either value between 1 and 65535
+            or <literal>auto</literal>. If <literal>auto</literal> is specified,
+            the port is automatically generated based on interface name.
+            Defaults to <literal>auto</literal>.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>FwMark</varname></term>
+        <listitem>
+          <para>Sets a firewall mark on outgoing wireguard packets from this interface.</para>
+        </listitem>
+      </varlistentry>
+    </variablelist>
+  </refsect1>
+
+  <refsect1>
+    <title>[WireGuardPeer] Section Options</title>
+
+    <para>The <literal>[WireGuardPeer]</literal> section accepts the following
+    key:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>PublicKey=</varname></term>
+        <listitem>
+          <para>Sets a base64 public key calculated by <command>wg pubkey</command> from a
+              private key, and usually transmitted out of band to the author
+              of the configuration file. This option is mandatory for this section.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>PresharedKey=</varname></term>
+        <listitem>
+          <para>Optional preshared key for the interface. It can be generated
+            by the <command>wg genpsk</command> command. This option adds an
+            additional layer of symmetric-key cryptography to be mixed into the
+            already existing public-key cryptography, for post-quantum
+            resistance.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>AllowedIPs=</varname></term>
```

Comment:
hmm, "IP" is short for "Internet Protocol", not for "IP Address", i.e. this setting's name isn't entirely correct.

Also, I think it would make sense to copy the name from the per-service `IPAddressAllow=` setting, as this is conceptually the same thing. Hence, please call this IPAddressAllow=

Reply:
You also wrote:

> but of course i have no expertise in wireguard.

And here it shows. AllowedIPs is a term from the WireGuard paper, from all current WireGuard tools, from every piece of WireGuard documentation, from every WireGuard man page, from every WireGuard conference talk, from every WireGuard blog post, from every WireGuard screencast, from every WireGuard training session. It's also not exactly conceptually similar to what's in systemd right now. Its name must not be changed. The time to bikeshed over the name was a very long time ago, and at this point, it's a term for a conceptually specific idea. It's a proper noun in its own right. Like it or not, it is what it is. @Mic92 - please don't change this.